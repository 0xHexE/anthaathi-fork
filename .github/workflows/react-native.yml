name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
