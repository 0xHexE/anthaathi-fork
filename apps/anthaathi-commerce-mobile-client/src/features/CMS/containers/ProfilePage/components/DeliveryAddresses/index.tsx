import {Pressable, View} from 'react-native';
import React from 'react';
import {Button, Card, Divider, Text} from 'react-native-paper';
import {useIntl} from 'react-intl';
import {ProfilePageComponentType} from '../../../../types/common';

type AddressProps = {
  firstName?: string;
  lastName?: string;
  companyName?: string;
  country: string;
  address: string;
  apartment: string;
  city: string;
  landmark: string;
  postalCode: number;
};

export interface DeliveryAddressesProps {
  title: string;
  handlePress?: () => void;
  userAddress: AddressProps[];
}

const DeliveryAddresses = (props: DeliveryAddressesProps) => {
  const intl = useIntl();
  return (
    <View style={{marginHorizontal: 10}} testID="deliveryAddresses">
      <View
        style={{
          flexDirection: 'row',
          justifyContent: 'space-between',
          alignItems: 'center',
        }}>
        <Text style={{color: '#364A15', fontSize: 16, fontWeight: '600'}}>
          {props.title}
        </Text>
        <Button
          testID="addNewAddress"
          mode="text"
          labelStyle={{color: '#008D3E'}}
          onPress={props.handlePress}>
          {intl.formatMessage({defaultMessage: 'Add New'})}
        </Button>
      </View>
      <Card
        style={{
          borderColor: '#E3E2E7',
          borderWidth: 1,
          borderRadius: 4,
          marginVertical: 2,
        }}
        testID="personalInformation">
        <Card.Content>
          {props.userAddress.map((data, index) => (
            <AddressRenderer key={index} data={data} />
          ))}
        </Card.Content>
      </Card>
    </View>
  );
};

const AddressRenderer = ({
  data,
  handlePress,
}: {
  data: AddressProps;
  handlePress?: () => void;
}) => {
  const intl = useIntl();
  return (
    <>
      <View
        testID="addressComponent"
        style={{
          flexDirection: 'row',
          justifyContent: 'flex-start',
          alignItems: 'flex-start',
          marginVertical: 5,
        }}>
        <View style={{width: '80%'}}>
          <Text
            testID="addressText1"
            style={{fontSize: 14, color: '#364A15', fontWeight: '600'}}>
            {data.apartment +
              ', ' +
              data.address +
              ', ' +
              data.city +
              ', ' +
              data.country}
          </Text>

          <Text style={{fontSize: 14, color: '#808080', fontWeight: '500'}}>
            {'Landmark: ' + data.landmark}
          </Text>
        </View>
        <Pressable
          testID="addressChangeButtonPress"
          onPress={handlePress}
          style={{width: '20%', alignItems: 'center'}}>
          <Text
            testID="addressChangeButtonTitle"
            style={{color: '#008D3E', fontSize: 14, fontWeight: '500'}}>
            {intl.formatMessage({defaultMessage: 'Edit'})}
          </Text>
        </Pressable>
      </View>
      <Divider style={{marginVertical: 4}} />
    </>
  );
};

export default DeliveryAddresses;

export const DeliveryAddressesCMSInput = {
  _component: ProfilePageComponentType.DeliveryAddresses,
  component: DeliveryAddresses,
};
