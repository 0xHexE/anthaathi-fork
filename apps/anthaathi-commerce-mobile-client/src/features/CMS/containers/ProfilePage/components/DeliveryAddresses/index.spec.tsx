import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import DeliveryAddresses from '.';

describe('DeliveryAddresses', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddresses
            title="Delivery Address"
            userAddress={[
              {
                address: 'address',
                country: 'Country',
                city: 'city',
                apartment: 'apartment',
                landmark: 'Some great landmark',
                postalCode: 12345,
              },
              {
                address: 'address',
                country: 'Country',
                city: 'city',
                apartment: 'apartment',
                landmark: 'Some great landmark',
                postalCode: 12345,
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('deliveryAddresses')).toBeTruthy();
  });

  it('should call when we call tap handlePress', function () {
    const onpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddresses
            title="Delivery Address"
            userAddress={[
              {
                address: 'address',
                country: 'Country',
                city: 'city',
                apartment: 'apartment',
                landmark: 'Some great landmark',
                postalCode: 12345,
              },
              {
                address: 'address',
                country: 'Country',
                city: 'city',
                apartment: 'apartment',
                landmark: 'Some great landmark',
                postalCode: 12345,
              },
            ]}
            handlePress={onpress}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('addNewAddress')!);
    expect(onpress).toBeCalledTimes(1);
  });

  it('should have DeliveryAddresses title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddresses
            title="Delivery Address"
            userAddress={[
              {
                address: 'address',
                country: 'Country',
                city: 'city',
                apartment: 'apartment',
                landmark: 'Some great landmark',
                postalCode: 12345,
              },
              {
                address: 'address',
                country: 'Country',
                city: 'city',
                apartment: 'apartment',
                landmark: 'Some great landmark',
                postalCode: 12345,
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Delivery Address')).toBeTruthy();
    expect(temp.queryByText('Add New')).toBeTruthy();
  });

  it('should have number of Delivery Addresses', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddresses
            title="Delivery Address"
            userAddress={[
              {
                address: 'address',
                country: 'Country',
                city: 'city',
                apartment: 'apartment',
                landmark: 'Some great landmark',
                postalCode: 12345,
              },
              {
                address: 'address',
                country: 'Country',
                city: 'city',
                apartment: 'apartment',
                landmark: 'Some great landmark',
                postalCode: 12345,
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryAllByTestId('addressComponent').length).toBe(2);
  });

  it('should have button title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddresses
            title="Delivery Address"
            userAddress={[
              {
                address: 'address',
                country: 'Country',
                city: 'city',
                apartment: 'apartment',
                landmark: 'Some great landmark',
                postalCode: 12345,
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('addressChangeButtonTitle')?.children).toContain(
      'Edit',
    );
  });

  it('should have address details', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddresses
            title="Delivery Address"
            userAddress={[
              {
                address: 'address',
                country: 'country',
                city: 'city',
                apartment: 'apartment',
                landmark: 'Some great landmark',
                postalCode: 12345,
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('addressText1')?.children).toContain(
      'apartment, address, city, country',
    );
  });
});
