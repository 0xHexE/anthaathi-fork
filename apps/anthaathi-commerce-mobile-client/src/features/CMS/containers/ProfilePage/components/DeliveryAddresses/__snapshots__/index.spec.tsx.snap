// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryAddresses should render the item 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 10,
    }
  }
  testID="deliveryAddresses"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(28, 27, 31, 1)",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#364A15",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
    >
      Delivery Address
    </Text>
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 20,
                },
              ]
            }
            testID="addNewAddress"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 12,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#008D3E",
                      },
                    ],
                  ]
                }
              >
                Add New
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "borderColor": "#E3E2E7",
            "borderRadius": 4,
            "borderWidth": 1,
            "elevation": 1,
            "marginVertical": 2,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexGrow": 1,
              "flexShrink": 1,
            }
          }
          testID="personalInformation"
        >
          <View
            style={
              Array [
                Object {
                  "paddingHorizontal": 16,
                },
                Object {
                  "paddingVertical": 16,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginVertical": 5,
                }
              }
              testID="addressComponent"
            >
              <View
                style={
                  Object {
                    "width": "80%",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#364A15",
                        "fontSize": 14,
                        "fontWeight": "600",
                      },
                    ]
                  }
                  testID="addressText1"
                >
                  apartment, address, city, Country
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#808080",
                        "fontSize": 14,
                        "fontWeight": "500",
                      },
                    ]
                  }
                >
                  Landmark: Some great landmark
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "width": "20%",
                  }
                }
                testID="addressChangeButtonPress"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#008D3E",
                        "fontSize": 14,
                        "fontWeight": "500",
                      },
                    ]
                  }
                  testID="addressChangeButtonTitle"
                >
                  Edit
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(231, 224, 236, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  Object {
                    "marginVertical": 4,
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginVertical": 5,
                }
              }
              testID="addressComponent"
            >
              <View
                style={
                  Object {
                    "width": "80%",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#364A15",
                        "fontSize": 14,
                        "fontWeight": "600",
                      },
                    ]
                  }
                  testID="addressText1"
                >
                  apartment, address, city, Country
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#808080",
                        "fontSize": 14,
                        "fontWeight": "500",
                      },
                    ]
                  }
                >
                  Landmark: Some great landmark
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "width": "20%",
                  }
                }
                testID="addressChangeButtonPress"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#008D3E",
                        "fontSize": 14,
                        "fontWeight": "500",
                      },
                    ]
                  }
                  testID="addressChangeButtonTitle"
                >
                  Edit
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(231, 224, 236, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  Object {
                    "marginVertical": 4,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
